langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java

```diff
@@ -22,26 +23,30 @@
 import static dev.langchain4j.internal.Utils.*;
 import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
 import static io.weaviate.client.v1.data.replication.model.ConsistencyLevel.QUORUM;
+import io.weaviate.client.v1.schema.model.Property;
 import static java.util.Arrays.stream;
 import static java.util.Collections.emptyList;
 import static java.util.Collections.singletonList;
 import static java.util.stream.Collectors.joining;
 import static java.util.stream.Collectors.toList;
+import org.apache.commons.lang3.ArrayUtils;
 
 /**
  * Represents the <a href="https://weaviate.io/">Weaviate</a> vector database.
  * Current implementation assumes the cosine distance metric is used.
- * Does not support storing {@link dev.langchain4j.data.document.Metadata} yet.
  */
 public class WeaviateEmbeddingStore implements EmbeddingStore<TextSegment> {
 
     private static final String METADATA_TEXT_SEGMENT = "text";
     private static final String ADDITIONALS = "_additional";
+    private static final String METADATA = "_metadata";
+    private static final String NULL_VALUE = "null";
```

Comment:
```suggestion
    private static final String NULL_VALUE = "<null>";
```
To make it clearer that this is a `null` value and not a "null" string